src/NHibernate.Test/NHSpecificTest/NH2319/Fixture.cs

```diff
@@ -118,6 +150,35 @@ public void TestFilter()
 			}
 		}
 
+		[Test]
+		public void TestPlanCacheMiss()
+		{
+			var internalPlanCache = typeof(QueryPlanCache)
+				.GetField("planCache", BindingFlags.NonPublic | BindingFlags.Instance)
+				?.GetValue(Sfi.QueryPlanCache) as SoftLimitMRUCache;
+			Assert.That(internalPlanCache, Is.Not.Null,
+				$"Unable to find the internal query plan cache for clearing it, please adapt code to current {nameof(QueryPlanCache)} implementation.");
+
+			using (var spy = new LogSpy(typeof(QueryPlanCache)))
+			{
+				internalPlanCache.Clear();
+				ShouldBeAbleToFindChildrenByName();
+				AssertFilterPlanCacheMiss(spy);
+			}
+		}
+
+		private const string _filterPlanCacheMissLog = "unable to locate collection-filter query plan in cache";
+
+		private static void AssertFilterPlanCacheHit(LogSpy spy) =>
+			// Each query currently ask the cache two times, so asserting reuse requires to check cache has not been missed
+			// rather than only asserting it has been hit.
+			Assert.That(spy.GetWholeLog(),
+				Contains.Substring("located collection-filter query plan in cache (")
+				.And.Not.Contains(_filterPlanCacheMissLog));
```

Comment:
The "not contains" is not much robust: if the log message in cache is changed, this will defeat this assert silently. Thus the additional `TestPlanCacheMiss()` above.